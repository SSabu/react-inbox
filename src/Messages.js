import React, { Component } from 'react';
import Message from './Message.js';

class Messages extends Component {

  render() {
    return (
      <div>
        <div>
            { this.props.messages.map( (message) => { return <Message message={message} key={message.id} toggleProperty={this.props.toggleProperty} /> })}
        </div>
      </div>
    )
  }
}

export default Messages;
