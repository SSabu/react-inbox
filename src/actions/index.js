export const MESSAGES_RECEIVED = 'MESSAGES_RECEIVED'
export const TOGGLE_SELECTED = 'TOGGLE_SELECTED'
export const TOGGLE_STARRED = 'TOGGLE_STARRED'
export const MARK_READ = 'MARK_READ'
export const MARK_UNREAD = 'MARK_UNREAD'
export const APPLY_LABEL = 'APPLY_LABEL'
export const REMOVE_LABEL = 'REMOVE_LABEL'
export const TOGGLE_SELECT_ON = 'TOGGLE_SELECT_ON'
export const TOGGLE_SELECT_OFF = 'TOGGLE_SELECT_OFF'
export const DELETE_MESSAGE = 'DELETE_MESSAGE'
export const SEND_MESSAGE = 'SEND_MESSAGE'
export const GET_BODY = 'GET_BODY'

export function getMessages(message) {
  return (dispatch, getState, { Api }) => {
    return Api.fetchMessages()
      .then(msgObj => msgObj.messages)
      .then(messages => dispatch({
        type: MESSAGES_RECEIVED,
        messages
      }))
    }
  }

export function toggleSelected(messageId) {
  return (dispatch, getState) => {
    return(dispatch({
      type: TOGGLE_SELECTED,
      messageId
    }))
  }
}

export function toggleStarred(messageId, starStatus){

  let updateObj = {
    "messageIds": [messageId],
    "command":"star",
    "star":!starStatus
  }

  return (dispatch, getState, { Api }) => {
    return Api.updateApiState('PATCH', updateObj)
      .then(messages => dispatch({
        type: TOGGLE_STARRED,
        messageId
      }))
  }
}

export function markRead(messageId){

  let updateObj = {
      "messageIds": messageId,
      "command":"read",
      "read": true
  }

  return (dispatch, getState, { Api }) => {
    return Api.updateApiState('PATCH', updateObj)
      .then(messages => dispatch({
        type: MARK_READ,
        messageId
      }))
  }
}

export function markUnread(messageId){

  let updateObj = {
    "messageIds": messageId,
    "command":"read",
    "read":false
  }

  return (dispatch, getState, { Api }) => {
    return Api.updateApiState('PATCH', updateObj)
      .then(messages => dispatch({
        type: MARK_UNREAD,
        messageId
      }))
  }
}

export function applyLabel(messageId, newLabel) {

    let updateObj = {
      "messageIds": messageId,
      "command":"addLabel",
      "label": newLabel
    }

    return (dispatch, getState, { Api }) => {
      return Api.updateApiState('PATCH', updateObj)
      .then(messages => dispatch({
        type: APPLY_LABEL,
        messageId,
        newLabel
      }))
  }
}

export function removeLabel(messageId, oldLabel) {

  let updateObj = {
    "messageIds": messageId,
    "command":"removeLabel",
    "label": oldLabel
  }

  return (dispatch, getState, { Api }) => {
    return Api.updateApiState('PATCH', updateObj)
      .then(messages => dispatch({
        type: REMOVE_LABEL,
        messageId,
        oldLabel
      }))
  }
}

export function toggleSelectOn() {
  return (dispatch, getState) => {
    return(dispatch({
      type: TOGGLE_SELECT_ON
    }))
  }
}

export function toggleSelectOff() {
  return (dispatch, getState) => {
    return(dispatch({
      type: TOGGLE_SELECT_OFF
    }))
  }
}

export function deleteMessages(messageId){

  let updateObj = {
    "messageIds":messageId,
    "command":"delete"
  }

  return (dispatch, getState, { Api }) => {
    return Api.updateApiState('PATCH', updateObj)
      .then(messages => dispatch({
      type: DELETE_MESSAGE,
      messageId
    }))
  }
}

export function sendMessage(subject, body) {
  let updateObj = {
    "body": body,
    "subject": subject
  }

  return (dispatch, getState, { Api }) => {
    return Api.updateApiState('POST', updateObj)
      .then(response => response.json())
      .then(msg => {
        dispatch({
          type: SEND_MESSAGE,
          subject,
          body,
          msg
        })
      })
  }
}

export function getMessageBody(path) {
  return (dispatch, getState, { Api }) => {
    return Api.fetchMessageBody(path)
      .then(messageBody => {
        return dispatch({
          type: GET_BODY,
          messageBody
        })
      })
  }
}
