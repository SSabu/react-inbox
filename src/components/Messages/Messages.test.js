import React from 'react'
import { shallow, render } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Messages, mapStateToProps, mapDispatchToProps } from './Messages'
import { getMessages } from '../../actions'

describe('Messages', () => {
  it('should render without any props defined', () => {
    const component = shallow(
      <Messages messages={[]} />
    );
    expect(toJson(component)).toMatchSnapshot();
  })

  it('should render with props passed in', () => {
    const component = shallow(
      <Messages messages={[{
        "id": 3,
        "subject": "Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!",
        "read": false,
        "starred": true,
        "labels": ["dev"]
      }]} />
    );
    expect(toJson(component)).toMatchSnapshot();
  })

  


})
