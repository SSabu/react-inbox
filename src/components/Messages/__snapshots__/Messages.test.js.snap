// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages should render with props passed in 1`] = `
<div>
  <div>
    <Connect(Message)
      id={3}
      labels={
        Array [
          "dev",
        ]
      }
      match={
        Object {
          "messages": Array [
            Object {
              "id": 3,
              "labels": Array [
                "dev",
              ],
              "read": false,
              "starred": true,
              "subject": "Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!",
            },
          ],
        }
      }
      read={false}
      starred={true}
      subject="Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!"
    />
  </div>
</div>
`;

exports[`Messages should render without any props defined 1`] = `
<div>
  <div />
</div>
`;
